intake_esm/source.py

```diff
@@ -79,15 +79,18 @@ def _open_dataset(
     )
 
     # Support kerchunk datasets, setting the file object (fo) and urlpath
-    if data_format == 'reference':
+    if data_format == 'reference' and xarray_open_kwargs['engine'] == 'zarr':
         xarray_open_kwargs['backend_kwargs']['storage_options']['fo'] = urlpath
         xarray_open_kwargs['backend_kwargs']['consolidated'] = False
         urlpath = 'reference://'
 
-    if xarray_open_kwargs['engine'] in 'zarr' or data_format == 'opendap':
+    if xarray_open_kwargs['engine'] == 'zarr' or data_format == 'opendap':
         url = urlpath
     elif fsspec.utils.can_be_local(urlpath):
         url = fsspec.open_local(urlpath, **storage_options)
+    # Support kerchunk datasets opened with the kerchunk engine
```

Comment:
Any chance you could add a test here for this? Just to make sure it works with the updated workflow + any upcoming changes to kerchunk

Reply:
Yes, I will work on this.